### PR TITLE
Add mono models to huggingface model-zoo and incorporate into pipeline

### PR DESCRIPTION
**Pull request by MXueguang**

### CODE REVIEW

File: pygaggle/run/evaluate_passage_ranker.py [hunk 1]
```diff
@@ -77,12 +78,15 @@ def tokenizer_sane(cls, v: str, values, **kwargs):
...
 
 def construct_t5(options: PassageRankingEvaluationOptions) -> Reranker:
-    loader = CachedT5ModelLoader(options.model_name_or_path,
-                                 SETTINGS.cache_dir,
-                                 'ranker',
-                                 options.model_type,
-                                 SETTINGS.flush_cache)
     device = torch.device(options.device)
+
+    try:
```
Comment (rodrigonogueira4) [inline pygaggle/run/evaluate_passage_ranker.py:83]: Instead of try/except, should we pass a flag to distinguish PT from TF checkpoints?

**Comment by ronakice** [comment]
I feel like the model in the model-zoo is missing the vocab file?

File: pygaggle/run/evaluate_passage_ranker.py (hunk 1 above)
Comment (MXueguang) [reply pygaggle/run/evaluate_passage_ranker.py:83]: @rodrigonogueira4 

Yes, I feel a flag makes more sense. will add an flag `--from_tf` for `evaluate_passage_ranker.py`

Comment (MXueguang) [reply pygaggle/run/evaluate_passage_ranker.py:83]: @ronakice oh, I didn't notice there is a one change difference between my gcloud and local. Will remove that line.

Comment (ronakice) [reply pygaggle/run/evaluate_passage_ranker.py:83]: Huggingface's work-around is something like `bool("ckpt" in model_name_or_path)`

Comment (MXueguang) [reply pygaggle/run/evaluate_passage_ranker.py:83]: model_name_or_path is a directory may contains both "ckpt" and `pytorch_model.bin` ? https://huggingface.co/google/bert_uncased_L-10_H-128_A-2#list-files

Comment (rodrigonogueira4) [reply pygaggle/run/evaluate_passage_ranker.py:83]: To avoid problems with PT files containing `ckpt` in their name, I would just go with the flag.

**Comment by MXueguang** [comment]
> I feel like the model in the model-zoo is missing the vocab file?

Is this because we are using t5-base as tokenizer?

**Comment by ronakice** [comment]
https://huggingface.co/valhalla/t5-base-squad#list-files compare this with our list-files and it seems like our model is missing a few!

**Comment by MXueguang** [comment]
Currently in `construct_t5()`
- we load model by `T5ForConditionalGeneration.from_pretrained("castorini/monot5-base-msmarco")`.
- we load tokenizer by `AutoTokenizer.from_pretrained("t5-base")`
which from different models on HuggingFace.

I think the missed files are from tokenizer.(e.g. config file and  spiece.model or vocal file). The pretrained model itself doesn't require these files

```
2020-05-24 02:04:42 [INFO] configuration_utils: loading configuration file https://s3.amazona
ws.com/models.huggingface.co/bert/t5-base-config.json from cache 

2020-05-24 02:04:42 [INFO] tokenization_utils: loading file https://s3.amazonaws.com/models.huggingface.co/bert/t5-spiece.model from cache
```

Do we need to find a way to integrate the pretrained tokenizer files to our model files?

**Comment by MXueguang** [comment]
> I don't think we need `default='false'` if `action='store_true'`. Could you double-check?

Yes, it can be removed.

**Comment by MXueguang** [comment]
revalidated experiment result on latest update

monot5
581590290744_ pic_hd

monoBERT
591590294894_ pic_hd

**Comment by ronakice** [comment]
Great! We can merge this. Thanks for doing this @MXueguang

**Pull request merged**